scripts/lib/api/generateApiComponents.ts

```diff
@@ -350,3 +399,44 @@ export async function htmlSignatureToMd(
     .replace(/^`/, "")
     .replace(/`$/, "");
 }
+
+function getHeaderLevel($: CheerioAPI, $main: Cheerio<any>, $dl: Cheerio<any>) {
+  const priorHeaderLevel = getPriorHeaderLevel($, $main, $dl);
+  if (priorHeaderLevel) {
+    if (+priorHeaderLevel == 6) {
+      throw new Error("API component cannot set inexisting header: <h7>");
+    }
+
+    return Math.max(3, +priorHeaderLevel + 1);
```

Comment:
It'd be good to use a variable to set the `3` number since you use it twice, e.g. `minLevel = 3`. For the comment, suggested tweak:

// We don't allow the header to be h1 or h2 because it's too large design-wise for API components.
// We try to ensure that the API docs are set up so there is always at least an h2 above the API component,
// but this is not always the case, especially with historical API docs. That means that we sometimes
// jump from h1 to h3. That's bad, but the tradeoff we're making to avoid using h2 for API components.